transformer.cc

```diff
@@ -37,6 +37,12 @@ enum struct WireFormat {
   kV1,
 };
 
+  // TODO(apang): Comment why this struct is needed.
+struct TraversalResult {
```

Comment:
We need to track the handle count for an old to v1 transformation, because when writing a union as a xunion, we need to write the actual (present) handle count into the envelope header.

You mention that the caller knows the handle count; I'm quite sure how. Can you clarify at all? We do have max_handles in the JSON IR, but that's the _maximum_ number of handles, not the _actual_ number of handles that may be present in the message. I could just be mistaking something.

And this info isn't on SrcDst since each transformation function may need to create their own local TraversalResult structure so that they can do something with it; putting it on SrcDst would be global, which doesn't allow for this. Does that make sense? I'll update the comment to explain.